tests/components/awair/test_config_flow.py

```diff
@@ -82,46 +83,69 @@ async def test_no_devices_error(hass):
         assert result["reason"] == "no_devices_found"
 
 
-async def test_reauth(hass):
+async def test_reauth(hass: HomeAssistant) -> None:
     """Test reauth flow."""
-    with patch(
-        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
-    ), patch(
-        "homeassistant.components.awair.sensor.async_setup_entry",
-        return_value=True,
-    ):
-        mock_config = MockConfigEntry(domain=DOMAIN, unique_id=UNIQUE_ID, data=CONFIG)
-        mock_config.add_to_hass(hass)
-        hass.config_entries.async_update_entry(
-            mock_config, data={**CONFIG, CONF_ACCESS_TOKEN: "blah"}
+    mock_config = MockConfigEntry(domain=DOMAIN, unique_id=UNIQUE_ID, data=CONFIG)
+    mock_config.add_to_hass(hass)
+    hass.config_entries.async_update_entry(
+        mock_config, data={**CONFIG, CONF_ACCESS_TOKEN: "blah"}
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_REAUTH, "unique_id": UNIQUE_ID},
+        data=CONFIG,
```

Comment:
We should pass the existing config entry data here.
```suggestion
        data={**CONFIG, CONF_ACCESS_TOKEN: "blah"},
```